lib/cache.js

```diff
@@ -131,6 +162,54 @@ with --force.`)
     this.npm.output(`Index entries: ${stats.totalEntries}`)
     this.npm.output(`Finished in ${stats.runTime.total / 1000}s`)
   }
+
+  async searchCacheForPackage (pkg, ver) {
+    const cachePath = path.join(this.npm.cache, '_cacache')
+    const cacheList = await cacache.ls(cachePath)
+    const keyList = Object.keys(cacheList)
+    const output = []
+    if (pkg === undefined)
+      pkg = '(.*)'
+    const search = new RegExp(`^make-fetch-happen:request-cache:(.*)/${pkg}/-/(.*)-(.*).tgz`)
+    for (const key of keyList) {
+      if (search.test(key))
+        output.push(key)
+    }
+    return output
+  }
+
+  async list (args) {
+    let pkg, ver
+    if (args.length > 0)
+      [pkg, ver] = args[0].split('@')
```

Comment:
shouldn't this use https://www.npmjs.com/package/npm-package-arg? the current approach seems like it will break on scoped packages.

Reply:
Yup. There's a bunch of changes needed here. This was a first pass POC. My first time getting hands on the codebase. I should have made that clear in my PR. The logic largely needs to move to cacache, the regexp needs some work, and you're absolutely right about `npm-package-arg`.

Reply:
Now uses `npm-package-arg`